elastic/beats
# [Kubernetes Module] Add missing metrics for replicaset and pod owners

- Enhancement

## Proposed commit message

- WHAT: We add information regarding the owner of pod and replicaset assets that was missing from Kube State Metrics endpoint.
- WHY:  We disable the metadata enrichemnt for deployments and replicasets by default as per https://github.com/elastic/elastic-agent-autodiscover/pull/62. This means that our pods and replicasets wont have the needed information for the process that creates them. We plan to fix this by introducsing relevant owner fields in state_pod and state_replicaset metricsets

## Checklist




## How to test this PR locally

1. Build metricbeat as per instructions [here](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/_meta/test/docs/README.md)
  This will copy your new metricbeat executable into the kubernetes cluster you want to observe
3. Login to your Elastic Stack (where metricbeat of previous step sends data) and create a new Dataview for metricbeat-* indices.
4. Observe the new `kubernetes.replicaset.replicas.owner.*` and `kubernetes.pod.owner.*` fields



## Related issues

- Relates #https://github.com/elastic/elastic-agent-autodiscover/issues/31


## Screenshots
For Pod that owner is node:
![Screenshot 2023-10-05 at 11 45 41 AM](https://github.com/elastic/beats/assets/1685415/f04b531a-8824-40c3-9b8b-fabf74bf965e)

For Pod that owner is replicaset:

![Screenshot 2023-10-05 at 11 46 16 AM](https://github.com/elastic/beats/assets/1685415/6c066c8b-a2d2-4f65-858e-c259cca91911)

For replicaset that owner is deployment:
![Screenshot 2023-10-05 at 11 46 41 AM](https://github.com/elastic/beats/assets/1685415/8eb9cf74-f16f-4e85-ae80-92da41809de2)
